SignalServiceKit/src/Messages/OWSMessageSender.m

```diff
@@ -976,7 +976,7 @@ - (void)sendMessageToService:(TSOutgoingMessage *)message
                                                                      messages:deviceMessages
                                                                         relay:recipient.relay
                                                                     timeStamp:message.timestamp];
-    if (TSSocketManager.sharedManager.canMakeRequests) {
+    if (CurrentAppContext().isMainApp && TSSocketManager.sharedManager.canMakeRequests) {
```

Comment:
We don't want to send over the web socket in the share extension.

Reply:
Though I agree we don't want to send over the web socket in the share extension, I also don't want to sprinkle concerns around that should be centralized.

The existing implementation of `canMakeRequests` relies on socket state. Are you saying that the socket was **open** in the share extension? That seems like a bug.

Or if you did *not* see that the socket was open in the SAE, and this change is just an abundance of caution, then would it be better to centralize the check in `TSSocketManager#canMakeRequests` rather than at every call site?

Reply:
Good idea.  See: https://github.com/signalapp/Signal-iOS/pull/3495